src/AppBundle/Controller/BlogController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AppBundle\Controller;
+
+use AppBundle\Entity\Comment;
+use AppBundle\Entity\Post;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Cache;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+
+/**
+ * Controller used to manage blog contents in the public part of the site.
+ *
+ * @Route("/blog")
+ *
+ * @author Ryan Weaver <weaverryan@gmail.com>
+ * @author Javier Eguiluz <javier.eguiluz@gmail.com>
+ */
+
+class BlogController extends Controller
+{
+    /**
+     * @Route("/", name="blog_index", defaults={"page" = 1})
+     * @Route("/page/{page}", name="blog_index_paginated", requirements={"page" : "\d+"})
+     * @Cache(smaxage="10")
+     */
+    public function indexAction($page)
+    {
+        $query = $this->getDoctrine()->getRepository('AppBundle:Post')->queryLatest();
+
+        $paginator = $this->get('knp_paginator');
+        $posts = $paginator->paginate($query, $page, Post::NUM_ITEMS);
+        $posts->setUsedRoute('blog_index_paginated');
+
+        return $this->render('blog/index.html.twig', array('posts' => $posts));
+
+    }
+
+    /**
+     * @Route("/posts/{slug}", name="blog_post")
+     *
+     * NOTE: The $post controller argument is automatically injected by Symfony
+     * after performing a database query looking for a Post with the 'slug'
+     * value given in the route.
+     * See http://symfony.com/doc/current/bundles/SensioFrameworkExtraBundle/annotations/converters.html
+     */
+    public function postShowAction(Post $post)
+    {
+        return $this->render('blog/post_show.html.twig', array('post' => $post));
+    }
+
+    /**
```

Comment:
__[development][code_sniffer]__
* Doc comment for parameter "$request" missing
* Doc comment for parameter "$post" missing

<!-- Comment by [SideCI](https://sideci.com) -->